File: MISC/CoderVation/MyNameIs1.0/MyNameIs.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class MyNameIs {

    public static void main(String[] args) {

        String newName,trueName,query,result;
         Map name = new HashMap();
        Scanner sc=new Scanner(System.in);
        sc.useDelimiter(" |\\n");
        int n=sc.nextInt();
        int q=sc.nextInt();
        for(int i=0;i<n;i++){
            trueName=sc.next();
            newName=sc.next();
            name.put(newName,trueName);         //put or insert newName,TrueName in Map
        }
        for(int i=0;i<q;i++){
            query=sc.next();        
            if(name.get(query)!=null)           //find given new name in Map
                System.out.println(name.get(query));//print their real name
            else
                System.out.println("Name not found");

    }

}
}

File: MISC/CoderVation/MyNameIs1.0/MyNameIs.cpp
#include <stdio.h>
	#include <string.h>
	#include <math.h>
	#include <stdlib.h>

	int main() {
    	int n,q,i,pos;
    	scanf("%d %d",&n,&q);
    	char trueName[10000][101],newName[10000][101],query[101];
    	for(i=0;i<n;i++)
       		 scanf("%s %s",&trueName[i][0],&newName[i][0]);
    	while(q--){
        	pos=-1;
        	scanf("%s",query);
        	for(int i=0;i<n;i++)
            	if(strcmp(query,newName[i])==0){
            		pos=i;
                }
             if(pos!=-1) 
                printf("%s\n",trueName[pos]);
            else
             	printf("Name not found\n");
                
    	}
    	return 0;
	}